File: email-validation-3-python/REST/ev3_response.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class Error:
    Type: Optional[str] = None
    TypeCode: Optional[str] = None
    Desc: Optional[str] = None
    DescCode: Optional[str] = None
    Number: Optional[str] = None

    def __str__(self) -> str:
        return (f"Error: Type={self.Type}, TypeCode={self.TypeCode}, "
                f"Desc={self.Desc}, DescCode={self.DescCode}")


@dataclass
class ValidateEmailInfo:
    Score: Optional[str] = None
    IsDeliverable: Optional[str] = None
    EmailAddressIn: Optional[str] = None
    EmailAddressOut: Optional[str] = None
    EmailCorrected: Optional[str] = None
    Box: Optional[str] = None
    Domain: Optional[str] = None
    TopLevelDomain: Optional[str] = None
    TopLevelDomainDescription: Optional[str] = None
    IsSMTPServerGood: Optional[str] = None
    IsCatchAllDomain: Optional[str] = None
    IsSMTPMailBoxGood: Optional[str] = None
    WarningCodes: Optional[str] = None
    WarningDescriptions: Optional[str] = None
    NotesCodes: Optional[str] = None
    NotesDescriptions: Optional[str] = None
    Error: Optional['Error'] = None

    def __str__(self) -> str:
        error = str(self.Error) if self.Error else "None"
        return (f"ValidateEmailInfo: Score={self.Score}, IsDeliverable={self.IsDeliverable}, "
                f"EmailAddressIn={self.EmailAddressIn}, EmailAddressOut={self.EmailAddressOut}, "
                f"EmailCorrected={self.EmailCorrected}, Box={self.Box}, Domain={self.Domain}, "
                f"TopLevelDomain={self.TopLevelDomain}, TopLevelDomainDescription={self.TopLevelDomainDescription}, "
                f"IsSMTPServerGood={self.IsSMTPServerGood}, IsCatchAllDomain={self.IsCatchAllDomain}, "
                f"IsSMTPMailBoxGood={self.IsSMTPMailBoxGood}, WarningCodes={self.WarningCodes}, "
                f"WarningDescriptions={self.WarningDescriptions}, NotesCodes={self.NotesCodes}, "
                f"NotesDescriptions={self.NotesDescriptions}, Error={error}")


@dataclass
class EV3Response:
    LicenseKey: Optional[str] = None
    EmailAddress: Optional[str] = None
    AllowCorrections: Optional[str] = None
    ValidateEmailInfo: Optional['ValidateEmailInfo'] = None
    Error: Optional['Error'] = None

    def __str__(self) -> str:
        validate_email_info = str(self.ValidateEmailInfo) if self.ValidateEmailInfo else "None"
        error = str(self.Error) if self.Error else "None"
        return (f"EV3Response: LicenseKey={self.LicenseKey}, EmailAddress={self.EmailAddress}, "
                f"AllowCorrections={self.AllowCorrections}, ValidateEmailInfo={validate_email_info}, Error={error}")

File: email-validation-3-nodejs/REST/ev3_response.js
export class ErrorModel {
    constructor(data = {}) {
        this.Type = data.Type;
        this.TypeCode = data.TypeCode;
        this.Desc = data.Desc;
        this.DescCode = data.DescCode;
        this.Number = data.Number;
    }

    toString() {
        return `Error: Type = ${this.Type}, TypeCode = ${this.TypeCode}, Desc = ${this.Desc}, DescCode = ${this.DescCode}`;
    }
}

export class ValidateEmailInfo {
    constructor(data = {}) {
        this.Score = data.Score;
        this.IsDeliverable = data.IsDeliverable;
        this.EmailAddressIn = data.EmailAddressIn;
        this.EmailAddressOut = data.EmailAddressOut;
        this.EmailCorrected = data.EmailCorrected;
        this.Box = data.Box;
        this.Domain = data.Domain;
        this.TopLevelDomain = data.TopLevelDomain;
        this.TopLevelDomainDescription = data.TopLevelDomainDescription;
        this.IsSMTPServerGood = data.IsSMTPServerGood;
        this.IsCatchAllDomain = data.IsCatchAllDomain;
        this.IsSMTPMailBoxGood = data.IsSMTPMailBoxGood;
        this.WarningCodes = data.WarningCodes;
        this.WarningDescriptions = data.WarningDescriptions;
        this.NotesCodes = data.NotesCodes;
        this.NotesDescriptions = data.NotesDescriptions;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        const error = this.Error ? this.Error.toString() : "None";
        return `ValidateEmailInfo: Score = ${this.Score}, IsDeliverable = ${this.IsDeliverable}, ` +
            `EmailAddressIn = ${this.EmailAddressIn}, EmailAddressOut = ${this.EmailAddressOut}, ` +
            `EmailCorrected = ${this.EmailCorrected}, Box = ${this.Box}, Domain = ${this.Domain}, ` +
            `TopLevelDomain = ${this.TopLevelDomain}, TopLevelDomainDescription = ${this.TopLevelDomainDescription}, ` +
            `IsSMTPServerGood = ${this.IsSMTPServerGood}, IsCatchAllDomain = ${this.IsCatchAllDomain}, ` +
            `IsSMTPMailBoxGood = ${this.IsSMTPMailBoxGood}, WarningCodes = ${this.WarningCodes}, ` +
            `WarningDescriptions = ${this.WarningDescriptions}, NotesCodes = ${this.NotesCodes}, ` +
            `NotesDescriptions = ${this.NotesDescriptions}, Error = ${error}`;
    }
}

export class EV3Response {
    constructor(data = {}) {
        this.LicenseKey = data.LicenseKey;
        this.EmailAddress = data.EmailAddress;
        this.AllowCorrections = data.AllowCorrections;
        this.ValidateEmailInfo = data.ValidateEmailInfo ? new ValidateEmailInfo(data.ValidateEmailInfo) : null;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        const validateEmailInfo = this.ValidateEmailInfo ? this.ValidateEmailInfo.toString() : "None";
        const error = this.Error ? this.Error.toString() : "None";
        return `EV3Response: LicenseKey = ${this.LicenseKey}, EmailAddress = ${this.EmailAddress}, ` +
            `AllowCorrections = ${this.AllowCorrections}, ValidateEmailInfo = ${validateEmailInfo}, Error = ${error}`;
    }
}

export default EV3Response;